server/sonar-server/src/main/java/org/sonar/server/license/ws/ListAction.java

```diff
@@ -0,0 +1,188 @@
+    context.createAction(ACTION_LIST)
+      .setDescription("List licenses settings.<br>" +
+        "Requires Administer System' permission")
+      .setResponseExample(getClass().getResource("list-example.json"))
+      .setSince("6.1")
+      .setInternal(true)
+      .setHandler(this);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    userSession.checkPermission(SYSTEM_ADMIN);
+
+    DbSession dbSession = dbClient.openSession(true);
+    try {
+      writeProtobuf(doHandle(dbSession), request, response);
+    } finally {
+      dbClient.closeSession(dbSession);
+    }
+  }
+
+  private ListWsResponse doHandle(DbSession dbSession) {
+    Set<String> settingsKeys = definitions.getAll().stream()
+      .filter(definition -> definition.type().equals(PropertyType.LICENSE))
+      .map(PropertyDefinition::key)
+      .collect(Collectors.toSet());
+    settingsKeys.add(PERMANENT_SERVER_ID);
+    List<PropertyDto> properties = dbClient.propertiesDao().selectGlobalPropertiesByKeys(dbSession, settingsKeys);
+    return new ListResponseBuilder(settingsKeys, properties).build();
+  }
+
+  private static class ListResponseBuilder {
+    private final Optional<String> serverId;
+    private final Map<String, PropertyDto> licenseSettingsByKey;
+    private final Set<String> licenseSettingsKeys;
+
+    ListResponseBuilder(Set<String> licenseSettingsKeys, List<PropertyDto> properties) {
+      this.serverId = getServerId(properties);
+      this.licenseSettingsKeys = licenseSettingsKeys.stream()
+        .filter(key -> !key.equals(PERMANENT_SERVER_ID))
+        .collect(Collectors.toSet());
+      this.licenseSettingsByKey = properties.stream().collect(Collectors.toMap(PropertyDto::getKey, Function.identity()));
+    }
+
+    ListWsResponse build() {
+      ListWsResponse.Builder wsResponse = ListWsResponse.newBuilder();
+      licenseSettingsKeys.forEach(key -> wsResponse.addLicenses(buildLicense(key, licenseSettingsByKey.get(key))));
+      return wsResponse.build();
+    }
+
+    private Licenses.License buildLicense(String key, @Nullable PropertyDto setting) {
+      Licenses.License.Builder licenseBuilder = Licenses.License.newBuilder().setKey(key);
+      if (setting != null) {
+        License license = License.readBase64(setting.getValue());
+        licenseBuilder.setValue(setting.getValue());
+        setProduct(licenseBuilder, license, setting);
+        setOrganization(licenseBuilder, license);
+        setExpiration(licenseBuilder, license);
+        setServerId(licenseBuilder, license);
+        setType(licenseBuilder, license);
+        setAdditionalProperties(licenseBuilder, license);
+      }
+      return licenseBuilder.build();
+    }
+
+    private static void setProduct(Licenses.License.Builder licenseBuilder, License license, PropertyDto setting) {
+      String product = license.getProduct();
+      if (product != null) {
+        licenseBuilder.setProduct(product);
+      }
```

Comment:
3 uses cases : 
- Product exists and is contain in setting key -> Return only product
- Product exists and is not contain in setting key -> Return product and invalidProduct
- Product doesn't exist -> Return invalidProduct